### ISSUE TITLE
Nullable not respected by builder

### ISSUE THREAD

**Issue by tkruse** [immutables/immutables#545]
Using Immutables 2.4.2
Maybe related to #484 

A Class like this:
```
@Value.Immutable
public abstract class Bus extends Vehicle {

    @CheckForNull
    abstract String driverName();
}
```
yields the following Builder code:
```
  /**
     * Initializes the value for the {@link Bus#driverName() driverName} attribute.
     * @param driverName The value for driverName 
     * @return {@code this} builder for use in a chained invocation
     */
    public final Builder driverName(String driverName) {
      this.driverName = Objects.requireNonNull(driverName, "driverName");
      initBits &= ~INIT_BIT_DRIVER_NAME;
      return this;
    }
```
The Objects.requireNonNull raises a Findbugs warning from method from(Object object).
```
private void from(Object object) {
      if (object instanceof Bus) {
        Bus instance = (Bus) object;
        driverName(instance.driverName());
      }
    }
```

I am not sure what the fix should be. Either remove the nullcheck in the builder setter, or wrap the setter in from (Object object) in a nullcheck.

**Comment by elucash** [comment]
Please, use `@Nullable` and not `@CheckForNull`. `CheckForNull` annotation is not supported in this way.

**Comment by tkruse** [comment]
@elucash : The problem is that Findbugs ignores @Nullable in this case.
See 
    https://checkerframework.org/manual/#findbugs-nullable
    http://stackoverflow.com/questions/31157511/how-to-use-findbugs-checkfornull-nonnull-and-nullable-annotations-correctly

So in order to use Findbugs to warn about potential NullPointer dereferences, Immutables would have to better support @CheckForNull.

Note:
If Immutables does want to support @CheckForNull, I must warn you about inconsistent/confusing Findbugs behavior: Do not make any assumptions, but test what findbugs actually reports. As an example, Findbugs might warn on bad usages if you copy the @CheckForNull annotation to the accessor method, but might drop the warnings if you also copy @CheckForNull on the private member.

**Comment by elucash** [comment]
You can put both. `@Nullable` is used for Immutables to allow `null` and as a way of documenting code. Also, while nullable predates type annotations in java8, it is usually used as such, so `@Nullable` is merely part of the type rather than method or parameter.  `@CheckForNull` can be used in addition to please FindBugs.
If you'd like, you can contribute a PR to handle this with a single annotation `@CheckForNull`. While not difficult it is not trivial either as all the usages on the parameters in the generated code should still use `@Nullable` but method return types should use either `@Nullable` or `@CheckForNull` depending on how it was declared. There is similar handling for Nullable annotations which are not applicable to local variables.

**Comment by tkruse** [comment]
I believe putting both did not solve my problem (because Findbugs is funny that way), I can try to reproduce to be sure.

**Comment by elucash** [comment]
In any case, if you're facing FB issues reported in the generated code, you can add a jar with `@SuppressFBWarnings` annotation to classpath and a generated code will be annotated with this annotation.

**Comment by tkruse** [comment]
Hi, If I use both @CheckForNull and @javax.annotations.Nullable, Findbugs does not generate the desired warnings in the code.

An ugly workaround I found is to use @javax.annotations.CheckForNull combined with another Nullable annotation such as: @com.jetbrains.annotations.Nullable.

I prepared a project for you to reproduce the issue:
https://github.com/tkruse/beantest/tree/immutablesIssue
Use the 'immutablesIssue' branch, not master. To reproduce, run ```mvn verify``` or ```gradlew check```.
